src/permitInfo/calculatePermitInfo.ts

```diff
@@ -0,0 +1,154 @@
+
+  // Load existing permitInfo.json file for given chainId if it exists
+  try {
+    allPermitInfo = JSON.parse(readFileSync(permitInfoPath, 'utf8')) as Record<string, PermitInfo>
+  } catch (e) {
+    console.info(`File ${permitInfoPath} not found. It'll be created.`, e)
+  }
+
+  // Build provider instance
+  const provider = getProvider(chainId, rpcUrl)
+
+  // Load tokens info from a token list
+  const tokens = getTokens(chainId, tokenListPath)
+
+  // Create a list of promises to check all tokens
+  const promises = tokens.map((token) => {
+    const existingInfo = allPermitInfo[token.address.toLowerCase()]
+
+    return fetchPermitInfo(chainId, provider, token, existingInfo)
+  })
+
+  // Await for all of them to complete
+  const results = await Promise.allSettled(promises)
+
+  // Iterate over each result
+  results.forEach((result) => {
+    // Ignore failed or the ones where the value is falsy
+    if (result.status === 'fulfilled' && result.value) {
+      const [address, permitInfo] = result.value
+
+      // Store result
+      allPermitInfo[address] = permitInfo
+    }
+  })
+
+  try {
+    writeFileSync(permitInfoPath, JSON.stringify(sortPermitInfo(allPermitInfo), undefined, 2))
+  } catch (e) {
+    console.error(`Failed to write file ${permitInfoPath}`, e)
+  }
+}
+
+function getProvider(chainId: number, rpcUrl: string | undefined): JsonRpcProvider {
+  const rpcEndpoint = rpcUrl ? rpcUrl : DEFAULT_RPC_URLS[chainId]
+
+  if (!rpcEndpoint) {
+    throw new Error(`No RPC found for network ${chainId}`)
+  }
+
+  if (!rpcUrl && (chainId === 1 || chainId === 5) && !process.env.INFURA_API_KEY) {
+    throw new Error(`INFURA_API_KEY is required`)
+  }
+
+  return new ethers.providers.JsonRpcProvider(rpcEndpoint)
+}
+
+const DEFAULT_RPC_URLS: Record<number, string> = {
+  1: 'https://mainnet.infura.io/v3/' + process.env.INFURA_API_KEY,
+  5: 'https://goerli.infura.io/v3/' + process.env.INFURA_API_KEY,
+  100: 'https://rpc.gnosischain.com',
+}
+
+type Token = {
+  address: string
+  name: string
+  chainId: number
+  symbol: string
+}
+
+function getTokens(chainId: number, tokenListPath: string | undefined): Array<Token> {
+  const filePath = tokenListPath
+    ? tokenListPath
+    : path.join(BASE_PATH, chainId === 5 ? 'CowSwapGoerli.json' : 'CowSwap.json')
+
+  return JSON.parse(readFileSync(filePath, 'utf-8')).tokens
+}
+
+async function fetchPermitInfo(
+  chainId: number,
+  provider: JsonRpcProvider,
+  token: Token,
+  existing: PermitInfo | undefined,
+): Promise<undefined | [string, PermitInfo]> {
+  if (existing !== undefined) {
+    console.info(`Token ${token.symbol}: already known, skipping`, existing)
+  } else if (token.chainId !== chainId) {
+    console.info(`Token ${token.symbol}: belongs to a different network (${token.chainId}), skipping`)
+  } else {
+    try {
+      const response = await getTokenPermitInfo({
+        chainId,
+        provider,
+        spender: '0xC92E8bdf79f0507f65a392b0ab4667716BFE0110',
```

Comment:
no hardcode

Reply:
Ok ok... 😅 
It just needs to be a valid address I think.